apis/python/src/tiledbsoma/tiledb_array.py

```diff
@@ -1,35 +1,49 @@
-from typing import List, Optional, Sequence, Tuple
+from typing import Optional, Sequence, Tuple, Type, TypeVar
 
 import pyarrow as pa
 import tiledb
+from somacore import options
 
 # This package's pybind11 code
 import tiledbsoma.libtiledbsoma as clib
+from tiledbsoma.options.soma_tiledb_context import SOMATileDBContext
 
-from .options import SOMATileDBContext
 from .tiledb_object import TileDBObject
 from .util_arrow import get_arrow_schema_from_tiledb_uri
+from .util_tiledb import ReadWriteHandle
 
+_Self = TypeVar("_Self", bound="TileDBArray")
 
-class TileDBArray(TileDBObject):
+
+class TileDBArray(TileDBObject[tiledb.Array]):
     """
     Wraps arrays from TileDB-Py by retaining a URI, options, etc.  Also serves as an abstraction layer to hide TileDB-specific details from the API, unless requested.
```

Comment:
comment slightly misleading, as it doesn't just wrap TileDB-Py.  I would ammend to "wraps TileDB"

Reply:
True but something I want to leave out of this.